ttnn/cpp/ttnn/operations/sliding_window/sliding_window.hpp

```diff
@@ -43,19 +43,21 @@ struct SlidingWindowConfig {
     uint32_pair_t pad_hw_;
     uint32_pair_t dilation_hw_;
 
+    uint32_t groups;
+
     // parallel configuration
     bool has_parallel_config_;
     uint32_t num_cores_nhw_;        // num cores along collapsed height nhw
     CoreRangeSet core_range_set_;   // active cores
 
     bool snap_to_tile_;
 
-    SlidingWindowConfig(uint32_t batch_size, uint32_t input_h, uint32_t input_w, uint32_t window_h, uint32_t window_w, uint32_t stride_h, uint32_t stride_w, uint32_t pad_h, uint32_t pad_w, uint32_t dilation_h = 1, uint32_t dilation_w = 1, uint32_t num_cores_nhw = 0, CoreRangeSet core_range = {{}}, bool snap_to_tile = false)
-        : batch_size_(batch_size), input_hw_(input_h, input_w), window_hw_(window_h, window_w), stride_hw_(stride_h, stride_w), pad_hw_(pad_h, pad_w), dilation_hw_(dilation_h, dilation_w), has_parallel_config_(false), num_cores_nhw_(num_cores_nhw), core_range_set_(core_range), snap_to_tile_(snap_to_tile) {
+    SlidingWindowConfig(uint32_t batch_size, uint32_t input_h, uint32_t input_w, uint32_t window_h, uint32_t window_w, uint32_t stride_h, uint32_t stride_w, uint32_t pad_h, uint32_t pad_w, uint32_t dilation_h = 1, uint32_t dilation_w = 1, uint32_t groups = 1, uint32_t num_cores_nhw = 0, CoreRangeSet core_range = {{}}, bool snap_to_tile = false)
```

Comment:
this looks really bad. 
I'd suggest to remove constructors and just have structure with default initialization as is and use designated initializers.
Maybe some static factory methods like create_silding_window_for_something(subset_of_params) might help too.

Reply:
This is a good idea

Reply:
Agreed. Removed the constructors and used designated initializers. Thank you for the suggestion.